Metrodr e2e

raghavendra-talur opened the pull request.

nirs commented on hack/ocm-minikube.sh:
```diff
@@ -944,16 +944,18 @@ hub_kubeconfig_file_create()
+ocm_minikube_deploy()
 {
+	hub_cluster_name=$1
+	spoke_cluster_names=$2
```
This creates a globals - is this used to pass the arguments to the functions below?

This is not a good direction - this code is has enough issues without globals.

nirs commented on hack/metrodr-setup.sh:
```diff
@@ -0,0 +1,229 @@
+        kubectl --context "${CEPHCLUSTER}" create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/crds.yaml
+        kubectl --context "${CEPHCLUSTER}" create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/operator.yaml
+	#echo "======Deploying cluster=======" && sleep 100
+	kubectl --context "${CEPHCLUSTER}" create -f "${basedir}/dev-rook-cluster.yaml"
```
Inconsistent tabs and spaces.

nirs commented on hack/metrodr-setup.sh:
```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
```
Other scripts use sh - bash is better but do we really want to maintain both sh and bash scripts?

I would remove both bash and sh and convert use python.

nirs commented on hack/metrodr-setup.sh:
```diff
@@ -0,0 +1,229 @@
+	fi
+}
+
+function externalCluster_setFSID() {
```
Inconsistent style, rest of code in hack seems to use lower_case.

nirs commented on hack/metrodr-setup.sh:
```diff
@@ -0,0 +1,229 @@
+
+function externalCluster_setFSID() {
+        declare rook_tools_pod
+        rook_tools_pod=$(kubectl --context "$1" -n "$2" get pods -o name | grep tools)
```
Using $1 and $2 make this code very hard to understand. Please do not add this in new code. This should be eliminated entirely in this project.

raghavendra-talur commented:
@nirs I have made some changes to this draft PR as it is something I am using for my current sync dr testing. I have addressed some of your comments as part of the changes but I don't intend to bring this PR out of draft status.

Please continue working on your changes using drenv. My goal is to have this working so that we can use it as a reference when your drenv code is merged to enable sync dr in that.
